packages/page-staking/src/Payouts/index.spec.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017-2021 @polkadot/app-staking authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { StakerState } from '@polkadot/react-hooks/types';
+
+import { mockHooks } from '../../test/hooks/defaults';
+import { PayoutsPage } from '../../test/pages/payoutsPage';
+
+jest.mock('@polkadot/react-hooks/useOwnEraRewards', () => ({
+  useOwnEraRewards: () => mockHooks.ownEraRewards
+}));
+
+describe('Payouts', () => {
+  let payoutsPage: PayoutsPage;
+  const someOwnedValidator: StakerState = {
+    controllerId: 'someId',
+    hexSessionIdNext: '0x1112',
+    hexSessionIdQueue: 'sth',
+    isLoading: false,
+    isOwnController: false,
+    isOwnStash: false,
+    isStashNominating: false,
+    isStashValidating: false,
+    sessionIds: [],
+    stashId: 'someStashId'
+  };
+
+  beforeEach(() => {
+    payoutsPage = new PayoutsPage();
+  });
+
+  describe('empty view', () => {
+    it('renders `My validators/stashes` toggle', async () => {
+      payoutsPage.renderPage();
+
+      await payoutsPage.expectText('My validators');
+      await payoutsPage.expectText('My stashes');
+    });
+
+    it('when no own validators, then `My validators` is disabled and not selected', async () => {
+      payoutsPage.renderPage();
+
+      await payoutsPage.expectMyValidators({ isDisabled: true, isSelected: false });
+    });
+
+    it('when some own validators, then `My validators` is enabled and selected', async () => {
+      payoutsPage.renderPage([someOwnedValidator]);
+
+      await payoutsPage.expectMyValidators({ isDisabled: false, isSelected: true });
+    });
+
+    it('when some own validators, but after accounts load and rerender, then `My validators` is selected',
+      async () => {
+        payoutsPage.renderPage();
+        await payoutsPage.expectMyValidators({ isDisabled: true, isSelected: false });
+        payoutsPage.renderPage([someOwnedValidator]);
+        await payoutsPage.expectMyValidators({ isDisabled: false, isSelected: true });
+      });
+
+    it('when loading own era rewards, then `My validators` and `My stashes` are disabled', async () => {
```

Comment:
I'm unsure if that's the best behaviour, to disable changes on e.g. long loading. What do you think?

Reply:
@jacogr - pls confirm if this makes sense. I'm not sure as I wasn't exposed to big data loads on those pages :)